Automattic/wp-calypso
# WPCOM: Migrate `wpcom.undocumented()` themes to `wpcom.req`

#### Changes proposed in this Pull Request

This PR migrates the `wpcom.undocumented()` theme management methods to `wpcom.req`. 

Part of the ongoing effort to get rid of `wpcom.undocumented()`.

#### Testing instructions

* Go to `/themes` as a logged out user.
* Verify you can view themes, filter, search, etc.
* Log in to WP.com.
* Go to `/themes/:site` where `:site` is a WP.com simple site.
* Verify themes are fetched correctly.
* Open a theme from the list and verify its details are still loaded properly.
* Try activating another theme and verify it still works.
* Go to `/themes/:site` where `:site` is a Jetpack site.
* Verify themes are fetched correctly.
* Open a theme from the list and verify its details are still loaded properly.
* Download a theme from WP.org and go to `/themes/upload/:site`
* Upload the zip and verify theme gets installed correctly.
* Activate the theme and verify activation works.
* Try activating another existing theme and verify it still works.
* Try deleting a theme and verify it works well.
* Start with a simple WP.com site, upgrade to a Business plan and go to `/themes/upload/:site`. Verify the process works the same way as on production.
* Verify all tests pass: `yarn run test-client client/state/themes/test/actions.js`
